app/static/js/script.js

```diff
@@ -267,16 +267,40 @@ function loadModalPopup(){
 
 
 const form = document.getElementById('form');
-const email = document.getElementById("email");
+const femail = document.getElementById("email");
+const fname = document.getElementById("name");
+const fmessage = document.getElementById("subject");
 const error = document.querySelector("small");
+
 form.addEventListener('submit', (e) => {
-	if(email.value !== email.value.toLowerCase()){
+	if(femail.value !== femail.value.toLowerCase()){
 		error.innerHTML = "Email should be in lower case";
 		error.style.display = 'block';
 		e.preventDefault();
 	}
 })
 
+// Local Storage 
+
+
+function saveUserData(){
+	let data = {
+		name: localStorage.setItem('Name', fname.value),
+		email: localStorage.setItem("Email", femail.value),
+		message: localStorage.setItem("Message", fmessage.value)
+	};
+	return data;
+}
+
+femail.value = localStorage.getItem("Email");
+fname.value = localStorage.getItem("Name");
+fmessage.value = localStorage.getItem("Message");
+
+saveUserData();
+
+fname.onchange = saveUserData;
+femail.onchange = saveUserData;
+fmessage.onchange = saveUserData;
```

Comment:
- [ ] There are 3 things expected from your form inputs:
     1. The contact form information is persistent to page reloads. ❌  [REQUIRED]
     2. Even when you don't submit, They should still persist the entered values❌  [REQUIRED]
     3. We should see the text auto-populated on our next visit❌ [REQUIRED]
     4. When they are empty we should get empty inputs on our next visit❌ [REQUIRED]
     5. It should save the data to local storage❌ [REQUIRED]
If you look at your app below you will notice that nothing gets saved on the right side (in the local storage). And on our next visit we get nothing.

| Your Form|
|----------|
|![kennanko1](https://user-images.githubusercontent.com/62938735/190627702-669598e1-e4ba-4a9d-ad50-a9870211d7ce.gif)|


Hints: 
 1. I don't think you need to call saveUserData on line 299
 2. One of the reasons. your code is probably. not working is because you did. not add brackets to saveUserData on lines 301, 302 and 303. You need the brackets here since you are not adding event listeners.
 3. Also I think you should review setting and getting items in `localstorage` because I don't think the code will still work from what I can see.

Reply:
Hello @fobadara 
I found that the problem was in another place in my code and not with the `usersData()` as this works just as intended as can be seen in the screen cast below 
[Screencast from 16-09-2022  4:03:31 WB.webm](https://user-images.githubusercontent.com/39535023/190647061-ad39baee-1d47-4e34-9058-6b899a6661c9.webm)